src/react/hoc/mutation-hoc.tsx

```diff
@@ -23,9 +23,9 @@ import { ApolloCache } from '../../core';
 export function withMutation<
   TProps extends TGraphQLVariables | {} = {},
   TData extends Record<string, any> = {},
-  TGraphQLVariables = {},
+  TGraphQLVariables extends OperationVariables = {},
   TChildProps = MutateProps<TData, TGraphQLVariables>,
-  TContext = DefaultContext,
+  TContext extends DefaultContext = DefaultContext,
```

Comment:
```suggestion
  TContext extends Record<string, any> = DefaultContext,
```

While these are technically the same types and this has no issue from the TypeScript compiler's perspective, I prefer to use a separate type assertion here. The reason is that `DefaultContext` is really meant to be a _default_, not necessarily the "type of this thing". `DefaultContext`. I like to think of `extends` as "a type that behaves as X". To me, its a bit funky to think of this as "a type that behaves like DefaultContext", really because of the "Default" in the name.

Thoughts?

Reply:
Sounds good to me 👍